src/internet_identity/src/stats/event_stats.rs

```diff
@@ -276,69 +293,130 @@ fn update_events_internal<M: Memory>(event: EventData, now: Timestamp, s: &mut S
-        s.event_data
-            .range(EventKey::min_key(prune_window_start)..=EventKey::max_key(prune_window_end))
-            .collect::<Vec<_>>()
-    } else {
-        // there is no event before `now` in the db, so the list of pruned events is empty
-        vec![]
-    };
-
+    let mut aggregations_db_wrapper = CountingAggregationsWrapper(&mut s.event_aggregations);
     // Update 24h aggregations
-    AGGREGATIONS.iter().for_each(|aggregation| {
-        update_aggregation(
-            |(_, data)| aggregation.process_event(AggregationWindow::Day, data),
-            current_key.clone(),
-            event.clone(),
-            &pruned_24h,
-            &mut CountingAggregationsWrapper(&mut s.event_aggregations),
-        );
-    });
+    update_aggregations(
+        &current_key,
+        &event,
+        &prune_events_24h(now, &current_key, &s.event_data),
+        AggregationWindow::Day,
+        &mut aggregations_db_wrapper,
+    );
 
     // This pruning _deletes_ the data older than 30 days. Do this after the 24h aggregation
     // otherwise the daily stats become inaccurate on the unlikely event that there is no activity
     // for 30 days.
-    let pruned_30d = prune_events(&mut s.event_data, now);
-
+    let pruned_30d = prune_events_30d(&mut s.event_data, now);
     // Update 30d aggregations
+    update_aggregations(
+        &current_key,
+        &event,
+        &pruned_30d,
+        AggregationWindow::Month,
+        &mut aggregations_db_wrapper,
+    );
+}
+
+/// Iterates over all aggregations and updates them based on the new event and the pruned events.
+fn update_aggregations<M: Memory>(
+    event_key: &EventKey,
+    event_data: &EventData,
+    pruned_events: &[(EventKey, EventData)],
+    window: AggregationWindow,
+    aggregations_db: &mut CountingAggregationsWrapper<M>,
+) {
     AGGREGATIONS.iter().for_each(|aggregation| {
         update_aggregation(
-            |(_, data)| aggregation.process_event(AggregationWindow::Month, data),
-            current_key.clone(),
-            event.clone(),
-            &pruned_30d,
-            &mut CountingAggregationsWrapper(&mut s.event_aggregations),
+            |(_, data)| aggregation.process_event(window.clone(), data),
+            event_key.clone(),
+            event_data.clone(),
+            pruned_events,
+            aggregations_db,
         );
     });
 }
 
-/// Adds an event to the event_data map and simultaneously removes events older than the retention period (30d).
+/// Collects events older than 24h that need to be removed from the 24h aggregations.
+/// Given events are kept for 30 days, the events are not deleted from the supplied `db`.
+/// Instead, this function simply updates the `event_stats_24h_pruning_start` state variable
+/// that denotes the first event that should be pruned in the next call.
+///
+/// Returns a vec of tuples of the pruned events and their timestamps, at most [MAX_EVENTS_TO_PRUNE].
+fn prune_events_24h<M: Memory>(
```

Comment:
This function doesn't actually prune any event, right? Could we rename it? `events_to_prune_24h` or `get_prune_events_24h`, does that make sense?

Reply:
Or even remove the "prune" altogether? It's more a list of events to update the aggregations rather than pruning them, right? `events_older_24h`?

Reply:
I renamed it to be clearer. 👍